src/custom/state/swap/hooks.tsx

```diff
@@ -257,6 +257,10 @@ export function useDerivedSwapInfo(): /* {
     chainId,
   })
 
+  // -- MOD --
+  const loadedUrlParams = useDefaultsFromURLSearch()
+  const loadedOutputCurrency = loadedUrlParams.OUTPUT.currencyId
```

Comment:
Why is this the solution? I mean, here the issue is that theres not input currency, right? (not the output)

Also, why do we limit ourselves to the selection coming from the URL params and not from the state of the selection itself.  We know if the user selected an input token or not, right?

Reply:
The issue is coming from this line https://github.com/cowprotocol/cowswap/blob/97c1d58a5225b6dbd632e773f61de4328e785625/src/custom/state/swap/hooks.tsx#L338 and this specific case where we click on buy COW so then there is no input currency. So we can detect this case by checking the query param for output currency and by adding this to the if condition it solves the issue

Reply:
but my point is that you check that we haven't loaded the OUTPUT currency, not the INPUT

my second point is, is this general enough? it seems like you are doing something very specific for this case instead of making sure that you control that we don't show this message if the input token is not selected (in general, not through query params)

Reply:
Maybe I'm missing your point, but we are already checking here if there is no INPUT currency https://github.com/cowprotocol/cowswap/blob/880344f200c852e5d98e76b5a51f2b0c03eb2391/src/custom/state/swap/hooks.tsx#L342, but we don't want this condition to match if there is output currency in the query params so that is why we are adding this additional `!loadedOutputCurrency` check.

Reply:
@anxolin I mean we could also change the test to this `if (!balanceIn && inputCurrency)`. Is that what you are having in mind?

Reply:
Yes, I see, u right. then I guess its only my other point. Why reading from the URL (`const loadedOutputCurrency = loadedUrlParams.OUTPUT.currencyId`), I mean, theres no selection, we should have a state for that right? no need to get this information for this specific case from the URL? 

maybe im missing sth, but why this case is not the same as: 
<img width="596" alt="image" src="https://user-images.githubusercontent.com/2352112/166916205-0f7242b0-078a-40f5-930d-0e05ba5b0d5e.png">

Reply:
Good point, maybe the best solution is then to check this `if (!balanceIn && inputCurrency)`, which means we show this message only when there is selected INPUT currency but there is NO balance loaded for that input currency which I think was intended logic to add this message in the first place.